### PR TITLE
OCPCLOUD-2565: Adds paused condition to Machine

### PR DESCRIPTION
**Pull request by theobarberbany**
This change adds a paused condition to the Machine controller, behind the `MachineAPIMigration` feature gate.

It sets the paused condition false when the Machine status has the `AuthoritativeAPI` set to MachineAPI, otherwise it sets it to true and stops reconciliation.

### CODE REVIEW

File: pkg/controller/machine/controller.go [hunk 1]
```diff
@@ -169,6 +180,37 @@ func (r *ReconcileMachine) Reconcile(ctx context.Context, request reconcile.Requ
 	// This must be a copy otherwise the referenced slice will be modified by later machine conditions changes.
 	originalConditions := conditions.DeepCopyConditions(m.Status.Conditions)
 
+	// Only enable MachineAPIMigration features if the featuregate is enabled.
+	if r.gate.Enabled(featuregate.Feature(openshiftfeatures.FeatureGateMachineAPIMigration)) {
+		// Check Status.AuthoritativeAPI
+		// If not MachineAPI. Set the paused condition true and return early.
+		if m.Status.AuthoritativeAPI != machinev1.MachineAuthorityMachineAPI {
```
Comment (JoelSpeed) [inline pkg/controller/machine/controller.go:187]: Where did we land on the handling of the empty string? I can't remember what we defined here?

Comment (theobarberbany) [reply pkg/controller/machine/controller.go:187]: Let me dig through the resolved comments on the original PR!

**Pull request merged**